src/stock.rs

```diff
@@ -276,6 +272,1129 @@ impl StockMessage {
+            .action_by_contact(contact)
+            .await
+            .msg()
+    }
+}
+
+#[derive(Debug)]
+pub(crate) enum MsgLocationDisabled {}
+
+impl MsgLocationDisabled {
+    /// Stock string: `Location streaming disabled.`.
+    pub async fn stock_str(context: &Context) -> Cow<'static, str> {
+        StockString::new(context)
+            .id(StockMessage::MsgLocationDisabled)
+            .await
+            .msg()
+    }
+}
+
+#[derive(Debug)]
+pub(crate) enum Location {}
+
+impl Location {
+    /// Stock string: `Location`.
+    pub async fn stock_str(context: &Context) -> Cow<'static, str> {
+        StockString::new(context)
+            .id(StockMessage::Location)
+            .await
+            .msg()
+    }
+}
+
+#[derive(Debug)]
+pub(crate) enum Sticker {}
+
+impl Sticker {
+    /// Stock string: `Sticker`.
+    pub async fn stock_str(context: &Context) -> Cow<'static, str> {
+        StockString::new(context)
+            .id(StockMessage::Sticker)
+            .await
+            .msg()
+    }
+}
+
+#[derive(Debug)]
+pub(crate) enum DeviceMessages {}
+
+impl DeviceMessages {
+    /// Stock string: `Device messages`.
+    pub async fn stock_str(context: &Context) -> Cow<'static, str> {
+        StockString::new(context)
+            .id(StockMessage::DeviceMessages)
+            .await
+            .msg()
+    }
+}
+
+#[derive(Debug)]
+pub(crate) enum SavedMessages {}
+
+impl SavedMessages {
+    /// Stock string: `Saved messages`.
+    pub async fn stock_str(context: &Context) -> Cow<'static, str> {
+        StockString::new(context)
+            .id(StockMessage::SavedMessages)
+            .await
+            .msg()
+    }
+}
+
+#[derive(Debug)]
+pub(crate) enum DeviceMessagesHint {}
+
+impl DeviceMessagesHint {
+    /// Stock string: `Messages in this chat are generated locally by...`.
+    pub async fn stock_str(context: &Context) -> Cow<'static, str> {
+        StockString::new(context)
+            .id(StockMessage::DeviceMessagesHint)
+            .await
+            .msg()
+    }
+}
+
+#[derive(Debug)]
+pub(crate) enum WelcomeMessage {}
+
+impl WelcomeMessage {
+    /// Stock string: `Welcome to Delta Chat! – ...`.
+    pub async fn stock_str(context: &Context) -> Cow<'static, str> {
+        StockString::new(context)
+            .id(StockMessage::WelcomeMessage)
+            .await
+            .msg()
+    }
+}
+
+#[derive(Debug)]
+pub(crate) enum UnknownSenderForChat {}
+
+impl UnknownSenderForChat {
+    /// Stock string: `Unknown sender for this chat. See 'info' for more details.`.
+    pub async fn stock_str(context: &Context) -> Cow<'static, str> {
+        StockString::new(context)
+            .id(StockMessage::UnknownSenderForChat)
+            .await
+            .msg()
+    }
+}
+
+#[derive(Debug)]
+pub(crate) enum SubjectForNewContact {}
+
+impl SubjectForNewContact {
+    /// Stock string: `Message from %1$s`.
+    // TODO: This can compute `self_name` itself instead of asking the caller to do this.
```

Comment:
Unfixed TODO

Reply:
I kind of did these TODOs on purpose.  I tried not to change more logic at the callers than necessary so did not want to change these behaviours.  But they could be nice things to do when someone is working on the calling site anyway.  So I left the TODOs to point someone to the fact that this could be nice.

Would you rather that I did these right away?  That I simply remove the TODOs?  Or leave as is?

Reply:
Ok to leave as is, I just though maybe you forgot to fix it.